### PR TITLE
Refactoring of XML extensions

### PR DESCRIPTION
**Pull request by MioRtia**
Signed-off-by: RALAMBOTIANA MIORA <miora.ralambotiana@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Refactoring of `CoordinatedReactiveControlXmlSerializer`, `TwoWindingsTransformerPhaseAngleClockXmlSerializer` and `ThreeWindingsTransformerPhaseAngleClockXmlSerializer` by using `AbstractExtensionXmlSerializer`

### CODE REVIEW

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/CoordinatedReactiveControlXmlSerializer.java [hunk 1]
```diff
@@ -7,39 +7,25 @@
 package com.powsybl.iidm.network.extensions;
 
 import com.google.auto.service.AutoService;
+import com.powsybl.commons.extensions.AbstractExtensionXmlSerializer;
 import com.powsybl.commons.extensions.ExtensionXmlSerializer;
```
Comment (agnesLeroy) [inline iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/CoordinatedReactiveControlXmlSerializer.java]: Do you still need this import?

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ThreeWindingsTransformerPhaseAngleClockXmlSerializer.java [hunk 2]
```diff
@@ -7,41 +7,27 @@
 package com.powsybl.iidm.network.extensions;
 
 import com.google.auto.service.AutoService;
+import com.powsybl.commons.extensions.AbstractExtensionXmlSerializer;
 import com.powsybl.commons.extensions.ExtensionXmlSerializer;
```
Comment (agnesLeroy) [inline iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ThreeWindingsTransformerPhaseAngleClockXmlSerializer.java]: Do you still need this import?

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/TwoWindingsTransformerPhaseAngleClockXmlSerializer.java [hunk 3]
```diff
@@ -7,41 +7,26 @@
 package com.powsybl.iidm.network.extensions;
 
 import com.google.auto.service.AutoService;
+import com.powsybl.commons.extensions.AbstractExtensionXmlSerializer;
 import com.powsybl.commons.extensions.ExtensionXmlSerializer;
```
Comment (agnesLeroy) [inline iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/TwoWindingsTransformerPhaseAngleClockXmlSerializer.java]: again, same question

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/CoordinatedReactiveControlXmlSerializer.java [hunk 4]
```diff
@@ -48,23 +34,8 @@ public void write(CoordinatedReactiveControl extension, XmlWriterContext context
     }
 
     @Override
-    public CoordinatedReactiveControl read(Generator extendable, XmlReaderContext context) throws XMLStreamException {
```
Comment (agnesLeroy) [inline iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/CoordinatedReactiveControlXmlSerializer.java]: why do you drop the XMLStreamException?

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/CoordinatedReactiveControlXmlSerializer.java (hunk 1 above)
Comment (MioRtia) [reply iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/CoordinatedReactiveControlXmlSerializer.java]: Yes, it is called in the `@AutoService` annotation

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ThreeWindingsTransformerPhaseAngleClockXmlSerializer.java (hunk 2 above)
Comment (MioRtia) [reply iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ThreeWindingsTransformerPhaseAngleClockXmlSerializer.java]: Same as above

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/TwoWindingsTransformerPhaseAngleClockXmlSerializer.java (hunk 3 above)
Comment (MioRtia) [reply iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/TwoWindingsTransformerPhaseAngleClockXmlSerializer.java]: Same as above

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/CoordinatedReactiveControlXmlSerializer.java (hunk 4 above)
Comment (MioRtia) [reply iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/CoordinatedReactiveControlXmlSerializer.java]: It is never thrown during this particular implementation, I suspect it was in the signature by default by the IDE template but it is not necessary here.

**Pull request merged**